trikot-streams/streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/processors/CombineLatestProcessor.kt

```diff
@@ -80,12 +80,13 @@ internal class CombineLatestProcessor<T>(
         private fun subscribeToCombinedPublishersIfNeeded() {
             if (hasSubscribed.compareAndSet(false, true)) {
                 val cancellableManager = cancellableManagerProvider.cancelPreviousAndCreate()
-                publishersResult.removeAll(publishersResult.value)
-
+                val newPublisherResultsList = mutableListOf<PublisherResult<T>>()
```

Comment:
If we're to never mutate the publishersResult values, why don't we use an AtomicReference (not a List) and assign the new reference?